wallet-core/src/transaction.rs

```diff
@@ -29,15 +29,16 @@ use execution_core::{
 };
 
 /// An unproven-transaction is nearly identical to a [`PhoenixTransaction`] with
-/// the only difference being that it carries a serialized [`TxCircuitVec`]
-/// instead of the proof bytes.
-/// This way it is possible to delegate the proof generation of the
-/// [`TxCircuitVec`] after the unproven transaction was created while at the
-/// same time ensuring non-malleability of the transaction, as the transaction's
-/// payload-hash is part of the public inputs of the circuit.
-/// Once the proof is generated from the [`TxCircuitVec`] bytes, it can
-/// replace the serialized circuit in the transaction by calling
-/// [`Transaction::replace_proof`].
+/// the only difference being that it carries a serialized
+/// [`execution_core::transfer::phoenix::TxCircuitVec`] instead of the proof
+/// bytes. This way it is possible to delegate the proof generation of the
+/// [`execution_core::transfer::phoenix::TxCircuitVec`] after the unproven
+/// transaction was created while at the same time ensuring non-malleability of
+/// the transaction, as the transaction's payload-hash is part of the public
+/// inputs of the circuit. Once the proof is generated from the
+/// [`execution_core::transfer::phoenix::TxCircuitVec`] bytes, it can replace
+/// the serialized circuit in the transaction by calling
+/// [`execution_core::transfer::phoenix::Transaction::set_proof`].
```

Comment:
This comment is actually not correct anymore and will be removed in https://github.com/dusk-network/rusk/pull/2289
To avoid merge conflicts it might be a good idea to revert these changes.